src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkaroundsInProcess.cs

```diff
@@ -17,6 +18,17 @@ namespace Roslyn.VisualStudio.IntegrationTests.InProcess
     [TestService]
     internal partial class WorkaroundsInProcess
     {
+        public async Task WaitForCompilationAvailableAsync(CancellationToken cancellationToken)
```

Comment:
Something to think about is whether this actually ensures the project exists in the first place from the project system, or whether this is just going to no-op. Might be a useful place to add extra logging.

Reply:
We already wait for project system and workspace.